File conflicts when building projects in parallel on Jenkins

I'm getting errors in CI where the `SetCloudBuildVariables` task is failing to write the `jenkins_build_number.txt`, presumably because other projects are writing it at the same time.  Should [Jenkins.WriteVersionFile](https://github.com/AArnott/Nerdbank.GitVersioning/blob/56aa4240de3c8eed70d95a9cf6db5679f6ac441c/src/NerdBank.GitVersioning/CloudBuildServices/Jenkins.cs#L50) be updated to use the same retry logic that was added in https://github.com/AArnott/Nerdbank.GitVersioning/commit/aeba28550fd60c4539f0f5731d5ddd0c1e5c2284?

> C:\Users\dotnet-bot\.nuget\packages\nerdbank.gitversioning\2.1.7\build\Nerdbank.GitVersioning.targets(117,5): error MSB4018: The "Nerdbank.GitVersioning.Tasks.SetCloudBuildVariables" task failed unexpectedly.
System.IO.IOException: The process cannot access the file 'D:\j\w\innerloop_Win---268ca7be\jenkins_build_number.txt' because it is being used by another process.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.StreamWriter.CreateFile(String path, Boolean append, Boolean checkHost)
   at System.IO.StreamWriter..ctor(String path, Boolean append, Encoding encoding, Int32 bufferSize, Boolean checkHost)
   at System.IO.File.InternalWriteAllText(String path, String contents, Encoding encoding, Boolean checkHost)
   at System.IO.File.WriteAllText(String path, String contents, Encoding encoding)
   at Nerdbank.GitVersioning.CloudBuildServices.Jenkins.WriteVersionFile(String buildNumber)
   at Nerdbank.GitVersioning.CloudBuildServices.Jenkins.SetCloudBuildNumber(String buildNumber, TextWriter stdout, TextWriter stderr)
   at Nerdbank.GitVersioning.Tasks.SetCloudBuildVariables.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext() [D:\j\w\innerloop_Win---268ca7be\src\Utilities.UnitTests\Microsoft.Build.Utilities.UnitTests.csproj]

--- a/src/NerdBank.GitVersioning/CloudBuildServices/Jenkins.cs
+++ b/src/NerdBank.GitVersioning/CloudBuildServices/Jenkins.cs
@@ -58,7 +58,7 @@
 
             var versionFilePath = Path.Combine(workspacePath, "jenkins_build_number.txt");
 
-            File.WriteAllText(versionFilePath, buildNumber, UTF8NoBOM);
+            Utilities.FileOperationWithRetry(() => File.WriteAllText(versionFilePath, buildNumber, UTF8NoBOM));
         }
     }
 }

--- a/src/NerdBank.GitVersioning/NerdBank.GitVersioning.csproj
+++ b/src/NerdBank.GitVersioning/NerdBank.GitVersioning.csproj
@@ -13,6 +13,6 @@
     <PackageReference Include="NuProj.Common" Version="0.11.14-beta" />
   </ItemGroup>
   <ItemGroup>
-    <Compile Include="..\Shared\Utilities.cs" LinkBase="Shared" />
+    <Compile Include="..\Shared\**\*.cs" LinkBase="Shared" />
   </ItemGroup>
 </Project>

--- a/src/NerdBank.GitVersioning/NerdBank.GitVersioning.csproj
+++ b/src/NerdBank.GitVersioning/NerdBank.GitVersioning.csproj
@@ -12,4 +12,7 @@
     <PackageReference Include="Nerdbank.GitVersioning.LKG" Version="1.6.20-beta-gfea83a8c9e" />
     <PackageReference Include="NuProj.Common" Version="0.11.14-beta" />
   </ItemGroup>
+  <ItemGroup>
+    <Compile Include="..\Shared\Utilities.cs" LinkBase="Shared" />
+  </ItemGroup>
 </Project>

--- a/src/Nerdbank.GitVersioning.Tasks/Nerdbank.GitVersioning.Tasks.csproj
+++ b/src/Nerdbank.GitVersioning.Tasks/Nerdbank.GitVersioning.Tasks.csproj
@@ -12,4 +12,7 @@
     <ProjectReference Include="..\MSBuildExtensionTask\MSBuildExtensionTask.csproj" />
     <ProjectReference Include="..\NerdBank.GitVersioning\NerdBank.GitVersioning.csproj" />
   </ItemGroup>
+  <ItemGroup>
+    <Compile Include="..\Shared\Utilities.cs" LinkBase="Shared" />
+  </ItemGroup>
 </Project>

--- a/src/Nerdbank.GitVersioning.Tasks/Nerdbank.GitVersioning.Tasks.csproj
+++ b/src/Nerdbank.GitVersioning.Tasks/Nerdbank.GitVersioning.Tasks.csproj
@@ -13,6 +13,6 @@
     <ProjectReference Include="..\NerdBank.GitVersioning\NerdBank.GitVersioning.csproj" />
   </ItemGroup>
   <ItemGroup>
-    <Compile Include="..\Shared\Utilities.cs" LinkBase="Shared" />
+    <Compile Include="..\Shared\**\*.cs" LinkBase="Shared" />
   </ItemGroup>
 </Project>

--- a/src/Shared/Utilities.cs
+++ b/src/Shared/Utilities.cs
@@ -4,7 +4,7 @@ using System.Threading;
 using System.Threading.Tasks;
 using Validation;
 
-namespace Nerdbank.GitVersioning.Tasks
+namespace Nerdbank.GitVersioning
 {
     internal static class Utilities
     {

